src/Bedrock.Framework.Experimental/Protocols/Tls/Hashs/HashProvider.cs

```diff
@@ -0,0 +1,13 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Bedrock.Framework.Experimental.Protocols.Tls.Hashs
+{
+    internal abstract class HashProvider
```

Comment:
Can we use any of the BCL types?

Reply:
So Tls during the handshake requires you to keep a running hash of the messages sent/received during the exchange. There is a BCL incremental hash function which is great but....

TLS requires you to grab the hash up until now at various points. This poses a problem as there is only a "gethashandreset" method where as I want to get the hash up to now and continue. Both BCrypt and OpenSsl support that scenario but the BCL does not. 

Without this you would have to buffer the messages during the handshake which is significantly worse and the interop for hash isn't a big deal.